### PR TITLE
Automate releases with goreleaser github action

### PR DESCRIPTION
**Pull request by jiparis**
This change adds a new github workflow to automate the releases on every vX.Y.Z pushed tag.
The expected result can be seen here: https://github.com/jiparis/pbvalidate/releases
Changelogs are also automatically computed.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Unshallow
+        run: git fetch --prune --unshallow
+      -
+        name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13.x
```
Comment (aruiz14) [inline .github/workflows/release.yml:23]: 1.14 is already out, update it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Unshallow
+        run: git fetch --prune --unshallow
```
Comment (aruiz14) [inline .github/workflows/release.yml:18]: Why is this needed?

Comment (jiparis) [reply .github/workflows/release.yml:18]: In https://github.com/goreleaser/goreleaser-action#usage there is this note:
```
IMPORTANT: note the Unshallow step. It is required for the changelog to work correctly.
```
So, it's for computing the changelogs properly.

**Comment by mkmik** [comment]
> The base branch requires all commits to be signed.

@jiparis please sign your commits with GPG

**Comment by jiparis** [comment]
> > The base branch requires all commits to be signed.
> 
> @jiparis please sign your commits with GPG

Done, I needed to re-push the changes.
